.husky/pre-commit

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+node "$(dirname "$0")/sort-dict/sort-dict.js" "$(dirname "$0")/../custom_dict.txt"
```

Comment:
```suggestion
node "$(dirname "$0")/sort-dict/sort-dict.js" "$(dirname "$0")/../custom_dict.txt"
git add $(dirname "$0")/../custom_dict.txt
```

Reply:
@tybritten I don't think the git add is necessary. I tested this locally and the list not sorted prior to committing and the resulting commit was sorted.

Reply:
@BrianMMcClain just had the opposite experience

Reply:
Hm, yeah that is strange. I tried it three times and each time it resulted in the unsorted addition to the dictionary being committed and _then_ sorted (which was weird since it's a pre-commit hook). 

So for example, if I make this change

```
diff --git a/custom_dict.txt b/custom_dict.txt
index d45c6f0..d9ebac9 100644
--- a/custom_dict.txt
+++ b/custom_dict.txt
@@ -5,6 +5,8 @@
 # where filename is relative to this configuration file
 # add entries to bottom of file, will be sorted on commit
+zzzzzz
+
 [0-9]*(a|b|c)
 [0-9]*(am|pm|AM|PM)
 [0-9]*(GB|MB|KB|Mi|m|m?s|k)
```

And then commit

```
➜  tanzu-dev-portal git:(sort-dict) ✗ git commit custom_dict.txt -m "Testing pre-commit hook"
[sort-dict 33e9e64] Testing pre-commit hook
 1 file changed, 2 insertions(+)
```

The commit is made

```
commit 33e9e64633f3e4b5a8ec02d7071379fe0805fe6e (HEAD -> sort-dict)
Author: Brian McClain <brianmmcclain@gmail.com>
Date:   Fri Feb 26 10:45:16 2021 -0500
    Testing pre-commit hook
```

But then I still have uncommitted changes with the newly sorted dictionary

```
diff --git a/custom_dict.txt b/custom_dict.txt
index d9ebac9..26bff1b 100644
--- a/custom_dict.txt
+++ b/custom_dict.txt
@@ -5,8 +5,6 @@
 # where filename is relative to this configuration file
 # add entries to bottom of file, will be sorted on commit
-zzzzzz
-
 [0-9]*(a|b|c)
 [0-9]*(am|pm|AM|PM)
 [0-9]*(GB|MB|KB|Mi|m|m?s|k)
@@ -1192,3 +1190,4 @@ ZDNet
 ZOP
 Zou
 Zuul
+zzzzzz
```

Could be PEBCAK and I'm overlooking something silly, though :)

Reply:
FWIW I just noticed the "add entries to the bottom of file" bit, but also seeing the same behavior doing that

Reply:
Weird, maybe I did something different yesterday or am misremembering. Just tried again and I'm now seeing the behavior you describe @BrianMMcClain. Let's keep the `git add`.